File: src/chipschallenge/blockreactions/NoSlipReaction.java

package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.BlockContainerFullException;


public abstract class NoSlipReaction extends BlockReaction {

    @Override
    public abstract void react(Block moving, Block standing) throws BlockContainerFullException;

    @Override
    public abstract boolean canMove(Block moving, Block standing);

    @Override
    public final boolean causesSlip(Block moving, Block standing) {
        return false;
    }

}

File: ChipsChallenge.Shared/Blockreactions/NoSlipReaction.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    using Moves = Move.Moves;

    public abstract class NoSlipReaction : BlockReaction
    {
        public abstract override void React(Block moving, Block standing);

        public abstract override bool canMove(Block moving, Block standing);

        public sealed override Moves? CausesSlip(Block moving, Block standing)
        {
            return null;
        }
    }
}